application/src/main/java/org/opentripplanner/transit/model/timetable/ScheduledTripTimes.java

```diff
@@ -366,8 +387,10 @@ private void validate() {
    * We really don't want those being used in routing. This method checks that all times are
    * increasing. The first stop arrival time and the last stops departure time is NOT checked -
    * these should be ignored by raptor.
+   *
+   * TODO: This should be make private as the builder should ensure the data consistency
```

Comment:
For other domain/builders data consistency is the responsibility of the domain class, not the builder. The reason is that we wanted all business rules for an entity to be in the main class and the builder to be dump. For the interpolation logic, we can extract the logic into a stategy, but it called from the builder.